classes/class-p4-search.php

```diff
@@ -189,26 +188,88 @@ protected function add_results_context( &$context ) {
 			];
 
 			// Sort filters alphabetically.
-			usort( $context['categories'], function( $a, $b ) {
+			uasort( $context['categories'], function( $a, $b ) {
 				return strcmp( $a['name'], $b['name'] );
 			} );
-			usort( $context['tags'], function( $a, $b ) {
+			uasort( $context['tags'], function( $a, $b ) {
 				return strcmp( $a['name'], $b['name'] );
 			});
 		}
 
+		/**
+		 * Handle form submit.
+		 *
+		 * @param array $context Associative array with the data to be passed to the view.
+		 *
+		 * @return bool Array if validation is ok, false if validation fails.
+		 */
+		public function handle_submit( &$context ) : bool {
+			if ( 'GET' === $_SERVER['REQUEST_METHOD'] ) {
+
+				// Handle submitted sort options.
+				$selected_sort = filter_input( INPUT_GET, 'orderby', FILTER_SANITIZE_STRING );
+				if ( ! in_array( $selected_sort, array_keys( $context['sort_options'] ), true ) ) {
+					$context['selected_sort'] = self::DEFAULT_SORT;
+				} else {
+					$context['selected_sort'] = $selected_sort;
+				}
+
+				// Handle submitted filter options.
+				if ( is_array( $_GET['f'] ) ) {
+					foreach ( $_GET['f'] as $type => $value ) {
+						foreach ( $value as $name => $id ) {
+
+							$id = filter_var( $id, FILTER_VALIDATE_INT );
+							if ( $this->validate( $id ) ) {
+								$this->filters[] = [
+									'type' => $type,
+									'id'   => $id,
+									'name' => $name,
+								];
+
+								switch ( $type ) {
+									case 'cat':
+										$context['categories'][ $id ]['checked'] = 'checked';
+										break;
+									case 'tag':
+										$context['tags'][ $id ]['checked'] = 'checked';
+										break;
+									case 'ptype':
+										$context['page_types'][ $id ]['checked'] = 'checked';
+										break;
+									case 'ctype':
+										$context['content_types'][ $id ]['checked'] = 'checked';
+										break;
+								}
+							}
+						}
+					}
+				}
+			}
+			return true;
+		}
+
+		/**
+		 * Validates the input.
+		 *
+		 * @param  int $id The id to be validated.
+		 *
+		 * @return bool True if validation is ok, false if validation fails.
+		 */
+		public function validate( $id ) : bool {
```

Comment:
Do you think we need a separate function to validate ID ? We can use `filter_var($int, FILTER_VALIDATE_INT))` 
}